src/vs/workbench/browser/parts/editor/multiEditorTabsControl.ts

```diff
@@ -160,6 +162,11 @@ export class MultiEditorTabsControl extends EditorTabsControl {
 
 		// React to decorations changing for our resource labels
 		this._register(this.tabResourceLabels.onDidChangeDecorations(() => this.doHandleDecorationsChange()));
+		this._register(this.configurationService.onDidChangeConfiguration(e => {
```

Comment:
We have `updateOptions` which is called when the options change. You can update the editor label there if you see this settings changed